### PR TITLE
Add icon loader implementation based on concept-fetch

### PR DESCRIPTION
**Pull request by pocmo**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by Amejia481** [comment]
There is a compilation error in Sample Browser

```
> Task :samples-browser:processGeckoBetaArmDebugResources
> Task :samples-browser:compileGeckoBetaArmDebugKotlin FAILED
e: /build/android-components/samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt: (53, 56): No value passed for parameter 'httpClient'
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':samples-browser:compileGeckoBetaArmDebugKotlin'.
> Compilation error. See log for more details

```

File: components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt [hunk 1]
```diff
@@ -28,21 +39,35 @@ import java.util.concurrent.Executors
  */
 class BrowserIcons(
     private val context: Context,
+    private val httpClient: Client,
```
Comment (Amejia481) [inline components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt]: nit: This looks like a breaking change, I think it should be added in the changelog.

```
> Task :samples-browser:processGeckoBetaArmDebugResources
> Task :samples-browser:compileGeckoBetaArmDebugKotlin FAILED
e: /build/android-components/samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt: (53, 56): No value passed for parameter 'httpClient'
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':samples-browser:compileGeckoBetaArmDebugKotlin'.
> Compilation error. See log for more details

```

**Comment by pocmo** [comment]
Fixed sample browser and added changelog entry. :)

**Comment by Amejia481** [comment]
👍

**Pull request merged**